apps/institution-service/src/controllers/institutionController.ts

```diff
@@ -277,31 +310,35 @@ export const getPaginatedInstitutions = async (req: Request, res: Response) => {
             "supports_history",
             "isActive",
           ],
-          where: whereAggregatorIntegrationConditions(req),
+          required: false,
           include: [
             {
               model: Aggregator,
               as: "aggregator",
               attributes: ["name", "id", "displayName", "logo"],
+              required: false,
             },
           ],
         },
       ],
-      distinct: true,
-      limit,
-      offset,
       order: [
         ["createdAt", "DESC"],
         ["name", "ASC"],
       ],
     });
 
+    const count = await Institution.count({
```

Comment:
Need updated integration tests for the changes in this file

Reply:
The previous integration tests already cover everything that changed in how the query is built. Do you think I should add additional tests here to duplicate the edge case tests which are covered in the E2E tests or what other tests should I add?

Reply:
I think we're out of whack with the testing trophy on this feature currently.

From our testing ADR: "The purpose of our end to end tests will be to validate that the frontend is working with the backend properly, and that apis are working together properly. We don't need to test every edge case in our end to end tests, because using MSW allows us to test those edge cases in our integration tests."

Every test we can write using an integration test should be done via an integration test. E2E tests are to add extra confidence, ensure that things are working together properly, and tests things that can't be tested via an integration test.

If we only wrote integration tests on this feature, then we could break the API because our integration tests don't know if express is setup properly. If we test all the filtering via integration tests, then we know if the correct parameters are passed, and express is setup properly, then it will continue to work. Adding an e2e test that really is just ensuring that the pieces are put together properly gives us the confidence we need.

If it were me I'd transfer a lot of the logic that the e2e tests are testing into integration tests, and just keep 1 or 2 e2e tests that test several filters all at once

Reply:
Do we still have 1 e2e test to give us confidence that express is setup correctly for filtering?

Reply:
Ok I added one back to the institutions.cy file